reactive/src/main/java/jmp/reactive/service/UserService.java

```diff
@@ -0,0 +1,55 @@
+package jmp.reactive.service;
+
+import jmp.reactive.client.UserClient;
+import jmp.reactive.model.User;
+import jmp.reactive.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.reactivestreams.Subscription;
+import org.springframework.context.SmartLifecycle;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.BaseSubscriber;
+
+@Service
+@RequiredArgsConstructor
+public class UserService implements SmartLifecycle {
+
+  private final UserRepository repository;
+  private final UserClient userClient;
+
+
+  @Override
+  public void start() {
+    userClient.getUsers()
+        .log()
+        .subscribe(
+            new BaseSubscriber<User>() {
+
+              int processed;
+              final int limit = 20;
+
+              @Override
+              protected void hookOnSubscribe(Subscription subscription) {
+                subscription.request(limit);
+              }
+
+              @Override
+              protected void hookOnNext(User value) {
+                if (++processed >= limit) {
+                  processed = 0;
+                  repository.save(value);
```

Comment:
You haven't subscribed to this publisher. Consequently, your db will be empty.

I get 404 Not Found error when I call API.
```shell
curl -v localhost:8080/api/v1/users?q=mojombo
*   Trying 127.0.0.1:8080...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /api/v1/users?q=mojombo HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/8.1.2
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< content-length: 0
< 
* Connection #0 to host localhost left intact
```